.github/workflows/holochain-build-and-test.yml

```diff
@@ -0,0 +1,90 @@
+name: "holochain build and test"
+
+on:
+  workflow_call:
+    inputs:
+      repo_path:
+        description: "a path to the holochain repository"
+        required: false
+        type: string
+        default: "."
+
+jobs:
+  "test":
+    strategy:
+      fail-fast: false
+      matrix:
+        cmd:
+          # ensure functionality on multiple platforms
+          - pkgs:
+              - holochain-tests-wasm
+              - holochain-tests-nextest
+              - holochain-tests-nextest-tx5
+              - holonix-tests-integration
+            extra_arg: "--override-input holochain ${{ inputs.repo_path }}"
```

Comment:
This tests the source code of `repo_path` agains the testing/build logic of the current branch, which is not necessarily the same.

I'd propose to remove the `--override-input` part, and instead execute `nix build ${{ inputs.repo_path }}" ...`, which ensures that the nix logic is used from the repo_path as well.

Reply:
Actually, what I just proposed would be quite verbose. Maybe better:
```
cd ${{ inputs.repo_path }}
nix build ...
```

Reply:
as per our conversation today, i understand why this can have advantages. i think it's too big of a leap for this PR as it'd require backporting many changes to the develop-0.1 branch and cut a new release there. let's have another conversation about how this approach would look like going forward and when we start the develop-0.2 branch.